packages/components/src/site-thumbnail/use-dominant-color.tsx

```diff
@@ -1,14 +1,22 @@
 import FastAverageColor, { FastAverageColorResult } from 'fast-average-color';
 import { useEffect, useState } from 'react';
 
+const colorMap = new Map< string, FastAverageColorResult >();
+
 export default function useDominantColor( url?: string ) {
-	const [ color, setColor ] = useState< FastAverageColorResult >();
+	const [ color, setColor ] = useState< FastAverageColorResult | undefined >(
+		( url && colorMap.get( url ) ) || undefined
+	);
 
 	useEffect( () => {
 		const fac = new FastAverageColor();
 
 		//get the dominant color of image from url
 		if ( url && url.length > 0 ) {
+			if ( colorMap.has( url ) ) {
+				return;
+			}
```

Comment:
We can do this check earlier, even before the `FastAverageColor` instance is created.